src/app/CompatEnumNames.h

```diff
@@ -0,0 +1,44 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * This file contains backwards-compatibility enum name definitions.  This file
+ * is meant to be included at the end of cluster-enums.h, after all the normal
+ * enum definitions are available.
+ */
+
+#pragma once
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+namespace OtaSoftwareUpdateProvider {
+using OTAApplyUpdateAction = ApplyUpdateActionEnum;
+using OTADownloadProtocol = DownloadProtocolEnum;
+using OTAQueryStatus = StatusEnum;
+} // namespace OtaSoftwareUpdateProvider
```

Comment:
Can you please add comments (written after initial push) for which PR introduced this and why?

Reply:
Sure thing.